### PR TITLE
Add desugaring examples

### PR DESCRIPTION
**Pull request by nathanKramer**
My goal here is to try to support [this PR](https://github.com/roc-lang/roc/pull/7133) by having some examples to link to that explain the desugaring of `!` and `?`.

I've basically used the examples from the tutorial added some notes and helpful links.

### CODE REVIEW

**Comment by Anton-4** [comment]
Thanks @nathanKramer :)
I further expanded and edited your examples so it fits with the style of the current examples.

**Comment by Anton-4** [comment]
Can one of you review @lukewilliamboswell or @smores56?

**Comment by smores56** [comment]
@Anton-4 I'll review tonight!

File: examples/DesugaringTry/main.roc [hunk 1]
```diff
@@ -0,0 +1,31 @@
+app [main] { pf: platform "https://github.com/roc-lang/basic-cli/releases/download/0.15.0/SlwdbJ-3GR7uBWQo6zlmYWNYOxnvo8r6YABXD-45UOw.tar.br" }
+
+import pf.Stdout
+
+main =
+    Stdout.line! (Inspect.toStr (parseNameAndYear "Alice was born in 1990"))
+    Stdout.line! (Inspect.toStr (parseNameAndYearTry "Alice was born in 1990"))
+
+### begin snippet question
```
Comment (nathanKramer) [inline examples/DesugaringTry/main.roc:9]:
```suggestion
### start snippet question
```

File: examples/DesugaringTry/main.roc [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+parseNameAndYear : Str -> Result { name : Str, birthYear : U16 } _
+parseNameAndYear = \str ->
+    { before: name, after: birthYearStr } = Str.splitFirst? str " was born in "
+    birthYear = Str.toU16? birthYearStr
+    Ok { name, birthYear }
+### end snippet question
+
+parseNameAndYearTry = \str ->
+    ### begin snippet try
```
Comment (nathanKramer) [inline examples/DesugaringTry/main.roc:18]:
```suggestion
    ### start snippet try
```

File: examples/DesugaringAwait/README.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+<details>
+  <summary>What's syntax sugar?</summary>
+  
+  Syntax within a programming language that is designed to make things easier 
+  to read or express. It allows developers to write code in a more concise, readable, or
+  convenient way without adding new functionality to the language itself.
+</details>
+
+Desugaring converts syntax sugar (like `x += 1`) into more fundamental operations (like `x = x + 1`).
```
Comment (lukewilliamboswell) [inline examples/DesugaringAwait/README.md:11]: I don't think this is blocking.. but this syntax may not be familiar to roc developers. Maybe there is a better example we could use here?

**Review by lukewilliamboswell — DISMISSED** [review]
This looks good. Thank you.

**Pull request merged**